R/spline_helpers.R

```diff
@@ -46,6 +46,14 @@ spline2_create <- function(x, nm = "pred", .fn = "bSpline", df = 3, complete_set
 
 spline_msg <- function(x) {
   x <- as.character(x)
+  # Error messages can contain brackets (e.g. "Error in if (df < 0) { : missing value")
+  # For glue string interpolation, the default open/close deliminators the
+  # brackets. cli_abort calls rlang's abort and that can't pass the arguments
+  # to change the delimiters but will ignore them if they are doubled. So we
+  # change "{" to "{{" (and also for close). Simultaneous substitution via
+  # `pattern = "(\\{)|(\\})"` produces poor results so we do them one at a time.
+  x <- gsub("(\\{)", "\\1\\1", x)
```

Comment:
Maybe this can be done in one shot?

Reply:
I'm perfectly fine to do it one at a time. we are throwing an error when this code is run so speed isn't an issue